server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -1208,12 +1208,64 @@ public void testRetrieveUnusedSegmentsUsingMultipleIntervalsAndNoLimit() throws
 
     final ImmutableList<DataSegment> actualUnusedSegments = retrieveUnusedSegmentsUsingMultipleIntervalsAndLimit(
         segments.stream().map(DataSegment::getInterval).collect(Collectors.toList()),
+        null,
         null
     );
     Assert.assertEquals(segments.size(), actualUnusedSegments.size());
     Assert.assertTrue(segments.containsAll(actualUnusedSegments));
   }
 
+  @Test
+  public void testRetrieveUnusedSegmentsUsingNoIntervalsAndNoLimit() throws IOException
+  {
+    final List<DataSegment> segments = createAndGetUsedYearSegments(1900, 2133);
+    markAllSegmentsUnused(new HashSet<>(segments));
+
+    final ImmutableList<DataSegment> actualUnusedSegments = retrieveUnusedSegmentsUsingMultipleIntervalsAndLimit(
+        ImmutableList.of(),
+        null,
+        null
+    );
+    Assert.assertEquals(segments.size(), actualUnusedSegments.size());
+    Assert.assertTrue(segments.containsAll(actualUnusedSegments));
+  }
+
+  @Test
+  public void testRetrieveUnusedSegmentsUsingNoIntervalsAndNoLimitAndOffset() throws IOException
+  {
+    final List<DataSegment> segments = createAndGetUsedYearSegments(2033, 2133);
+    markAllSegmentsUnused(new HashSet<>(segments));
+
+    int offset = 10;
+    List<DataSegment> expectedSegments = segments.stream().sorted((Comparator<DataSegment>) (o1, o2) -> {
+      long o1Start = o1.getInterval().getStartMillis();
+      long o2Start = o2.getInterval().getStartMillis();
+      if (o1Start < o2Start) {
+        return -1;
+      } else if (o1Start > o2Start) {
+        return 1;
+      } else {
+        long o1End = o1.getInterval().getEndMillis();
+        long o2End = o2.getInterval().getEndMillis();
+        if (o1End < o2End) {
+          return -1;
+        }
+        return o1End > o2End
+            ? 1
+            : 0;
+      }
+    })
+        .skip(offset)
+        .collect(Collectors.toList());
+    final ImmutableList<DataSegment> actualUnusedSegments = retrieveUnusedSegmentsUsingMultipleIntervalsAndLimit(
+        ImmutableList.of(),
+        null,
+        offset
+    );
+    Assert.assertEquals(expectedSegments.size(), actualUnusedSegments.size());
+    Assert.assertTrue(expectedSegments.containsAll(actualUnusedSegments));
+  }
+
```

Comment:
For test coverage, can we also please include new or extend existing tests for the following scenarios:
1. non-empty interval, non-null limit and non-null offset
2. multiple intervals, non-null limit and non-null offset
3. multiple intervals exceeding `MAX_INTERVALS_PER_BATCH`, non-null limit and non-null offset (the offset in this case would be ignored)

Reply:
added